### ISSUE TITLE
TabPFN for Bayesian optimization

### ISSUE THREAD

**Issue by songlei00** [PriorLabs/TabPFN#1116]
Hi TabPFN Team,

Thank you for your incredible work on TabPFN!

I have been studying the source code of [PFNs4BO](https://github.com/automl/PFNs4BO/tree/main) and noticed this repository for the TabPFN foundation model. Since this repo leverages a stronger version of PFN, I would love to use it to power my own Bayesian Optimization (BO) workflow.

I noticed that a differentiable version of the PFN regressor was recently released. This is extremely promising for BO, as it theoretically allows us to optimize the acquisition function via gradient descent. However, I am a bit unsure how to correctly invoke the forward pass to obtain the logits required for calculating the Expected Improvement (EI). **Since the standard predict() method adheres to the scikit-learn API, passing a PyTorch Tensor directly into it throws an error as it only supports scikit-learn compatible inputs (like NumPy arrays), which unfortunately breaks the PyTorch computation graph.**

Therefore, I need to find a way to compute the logits and criterion via a lower-level forward-pass call that natively supports Tensors and preserves gradients. Could you please provide a minimal working example on how to achieve something like the following?

``` python
import torch
# Assuming data_X requires gradient for acquisition optimization
data_X.requires_grad_() 

model = TabPFNRegressor(xxx)
model.fit_with_differentiable_input(train_X, train_y)

# We want to obtain logits and criterion while keeping the gradients intact back to data_X
# (Instead of using the sklearn-style predict() which throws errors for Tensors)
result = model.forward_like_call(data_X, output_type="full") 
logits: torch.Tensor = result["logits"]
criterion = result["criterion"]

# Compute EI and backpropagate to get gradients w.r.t data_X
ei_values = criterion.ei(logits, best_f=best_f, maximize=True)
ei_values.backward() 
# print(data_X.grad)
```

Any guidance or code snippets on how to properly set up this differentiable pipeline would be highly appreciated!

**Comment by klemens-floege** [comment]
@songlei00 thank you for opening an Issue and for your kind words; the [PFNs4BO](https://github.com/automl/PFNs4BO/tree/main) is very cool work but comes from a different codebase. We are also excited about BO but do not support it yet out of the box in our package. Give us a bit of time to look into this and we will get back to you.

**Comment by klemens-floege** [comment]
@songlei00 thank you for asking about BO, we took some time internally and implemented a BO version with prompt tuning in our extensions package, please see: https://github.com/PriorLabs/tabpfn-extensions/pull/343. I will close this issue here for now; feel free to expand on this BO work in tabpfn-extensions :)
